### PR TITLE
Add an ability to use scan queries via new YDB operator

### PR DESCRIPTION
**Pull request by vgvoleg**
Add an ability to use YDB scan queries via new YDB operator
https://ydb.tech/docs/en/concepts/scan_query

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py [hunk 1]
```diff
@@ -77,11 +88,13 @@ def wrapper(*args, **kwargs):
 class Cursor:
     def __init__(
         self,
+        driver: Union[ydb.Driver, ydb.aio.Driver],
```
Comment (ashb) [inline airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py:87]: This is in a `_vendor` file -- we shouldn't be making changes to it here. What version of the upstream file has these changes in?

**Review by ashb — DISMISSED** [review]
This makes changes to the vendored file that are not included in the upstream project https://github.com/ydb-platform/ydb-sqlalchemy/blob/main/ydb_sqlalchemy/dbapi/cursor.py -- we should submit the change there first please, or find a way to make this change that doesn't involve the vendored files diverging from the upstream project in terms of features.

File: airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py (hunk 1 above)
Comment (vgvoleg) [reply airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py:87]: This was added here https://github.com/ydb-platform/ydb-sqlalchemy/commit/91a13e059c3f8a135474e181b410af901a34ab0b

Comment (uzhastik) [reply airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py:87]: I would suggest to update readme https://github.com/apache/airflow/blob/360e727ef0ed9d151e343c497e20aa609e06e58e/airflow/providers/ydb/hooks/_vendor/readme.md

and fill in precise revision of vendor package

Comment (vgvoleg) [reply airflow/providers/ydb/hooks/_vendor/dbapi/cursor.py:87]: > I would suggest to update readme https://github.com/apache/airflow/blob/360e727ef0ed9d151e343c497e20aa609e06e58e/airflow/providers/ydb/hooks/_vendor/readme.md
> 
> and fill in precise revision of vendor package

done

**Review by uzhastik — APPROVED** [review]

**Comment by potiuk** [comment]
> This makes changes to the vendored file that are not included in the upstream project https://github.com/ydb-platform/ydb-sqlalchemy/blob/main/ydb_sqlalchemy/dbapi/cursor.py -- we should submit the change there first please, or find a way to make this change that doesn't involve the vendored files diverging from the upstream project in terms of features.

Quite agree with @ashb  on that one.

**Comment by vgvoleg** [comment]
> > This makes changes to the vendored file that are not included in the upstream project https://github.com/ydb-platform/ydb-sqlalchemy/blob/main/ydb_sqlalchemy/dbapi/cursor.py -- we should submit the change there first please, or find a way to make this change that doesn't involve the vendored files diverging from the upstream project in terms of features.
> 
> Quite agree with @ashb on that one.

These changes are already on vendor side, I've just updated the version of copied files. We still have no separate package to import directly without any issues, the problem is described here https://github.com/apache/airflow/blob/3bb6d02107493af64c4ed1b5fb4a6b8448c29945/airflow/providers/ydb/hooks/_vendor/readme.md

**Review by potiuk — APPROVED** [review]
Ah. OK then. LGTM.

**Comment by potiuk** [comment]
@ashb -> can you remove the "require changes" please?

**Comment by ashb** [comment]
I can't from my phone but please dismiss it if you can

**Pull request merged**